### PR TITLE
Card

### PR DESCRIPTION
**Pull request by AnuragSahu11**
Capture
Capture2

*Created all types of cards.
*Updated button
*I use width and min-width, height and min-height both so that if in case the content inside the element tries to overflow, the element will automatically adjust itself to contain it.

### CODE REVIEW

File: components/card.css [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    border-radius: 10px;
+    min-height: 430px;
+    height: min-content;
+    display: flex;
+    flex-wrap: wrap;
+    padding-bottom: 10px;
+    overflow: hidden;
+    position: relative;
+  }
+  .card-head {
```
Comment (ritik-patel05) [inline components/card.css:15]: A better name card-title or card-heading?

File: components/card.css [hunk 2]
```diff
@@ -0,0 +1,126 @@
+
+/* Basic card */
+
+.card {
+    width: 310px;
+    border-radius: 10px;
+    min-height: 430px;
+    height: min-content;
```
Comment (ritik-patel05) [inline components/card.css:8]: isn't it by default min-content? If it is redundant remove it.

File: components/card.css [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  .card-body {
+    padding-left: 24px;
+    padding-right: 24px;
+  }
+  
+  /* Card image */
+  
+  .card-image {
+    border: none;
+    height: min-content;
```
Comment (ritik-patel05) [inline components/card.css:28]: redundant?

File: components/card.css [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    width: 100%;
+    background-color: grey;
+  }
+  .card-image-img {
```
Comment (ritik-patel05) [inline components/card.css:32]: Better naming?
is-img?

File: components/card.css [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    width: 100%;
+  }
+  .card-body {
+    padding-left: 24px;
+    padding-right: 24px;
+  }
+  
+  /* Card image */
+  
+  .card-image {
```
Comment (ritik-patel05) [inline components/card.css:26]: just .card-img?

File: components/card.css [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+  
+  /* card horizontal */
+  
+  .card-horizontal{
+      padding: 8px;
+      justify-content: left;
+      align-items: center;
+      flex-direction: row;
+      display: flex;
+      min-width:700px;
```
Comment (ritik-patel05) [inline components/card.css:104]: have consistent spacing after colon.
At other places you have given one space after colon.

File: components/card.css [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+  /* card text only */
+
+  .card-text-only{
+      max-height: fit-content;
+      min-height: min-content;
+  }
+  
+
```
Comment (ritik-patel05) [inline components/card.css:126]: Showing no newline at the end of file. Better to add it.

**Review by ritik-patel05 — COMMENTED** [review]
- Use consistent code formatting, formatting is broken at several places. 
- Can create a style guide for spacing units that are commonly used or a set of spaces that you might use in the future. It will help you stay consistent in choosing from, and no need to change at every place you use that unit. Say no to choosing units randomly like 7, 13. Hope that makes sense. For example, you can look at my or @Hamdrive code for example.

File: components/card.css [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+      margin-left: auto;
+      margin-right: auto;
+    height: 100%;
+    display: block;
+  }
+  
+  /* card cross */
+  
+  .card-cross {
+      position: absolute;
```
Comment (Hamdrive) [inline components/card.css:42]: Can use helper class variables for clumping repeated properties like Position, top, right etc (under one class)

File: components/card.css [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+    color: var(--hover-red);
+  }
+  
+  /* card badge */
+  
+  .card-badge {
+    position: absolute;
+    font-size: 1.8rem;
+    top: 10px;
+    padding-left: 3px;
```
Comment (Hamdrive) [inline components/card.css:70]: Use short syntax if possible and is easier to read
`padding: 0 3px`

File: components/card.css [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+    padding-right: 3px;
+    border-radius: 3px;
+    background-color: var(--primary);
+    color: white;
```
Comment (Hamdrive) [inline components/card.css:74]: Use predefined stylesheet color variables

File: components/card.css [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+  .card-overlay{
+    opacity: 0.2;
+  }
+  .card-overlay-text{
+      position: absolute;
+      left: 0;
+      bottom: 0;
+      right: 0;
+      top: 0;
+      display: flex;
```
Comment (Hamdrive) [inline components/card.css:88]: Can store common flex properties in a helper class as well

File: utilities/utilities.css [hunk 12]
```diff
@@ -43,7 +43,7 @@
   font-size: 24px;
```
Comment (Hamdrive) [inline utilities/utilities.css:43]: Suggestion to use `rem` for anything larger than 12-16px

**Review by Hamdrive — COMMENTED** [review]
Good work done so far, some inconsistencies which @ritik-patel05 has pointed out that can be easily refactored.
Apart from those points, I have given suggestions that will help make your CSS overall easier to use.

**Comment by AnuragSahu11** [comment]
Thanks Ritik and Hamza I have made the changes suggested by both of you.

**Pull request merged**